lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -1205,6 +1202,72 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
         }
         return failure();
       });
+  patterns.onOp(
+      "GlobalMaxPool", 1,
+      [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value operand;
+        if (binder.tensorOperand(operand) ||
+            binder.tensorResultType(resultType))
+          return failure();
+
+        auto inputTensorType = operand.getType().cast<Torch::ValueTensorType>();
+        if (!inputTensorType || !inputTensorType.hasSizes()) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "Expected input type having sizes");
+        }
+        ArrayRef<int64_t> inputShape = inputTensorType.getSizes();
+        unsigned inputRank = inputShape.size();
+        if (!resultType || !resultType.hasSizes()) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "Expected result type having sizes");
+        }
+        ArrayRef<int64_t> resultShape = resultType.getSizes();
+        unsigned resultRank = resultShape.size();
+
+        SmallVector<Value> outputSize;
+        Value cstOne = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getI64IntegerAttr(1));
+        for (unsigned i = 2; i < resultRank; i++) {
+          if (resultShape[i] == Torch::kUnknownSize) {
+            outputSize.push_back(cstOne);
+          } else {
+            outputSize.push_back(rewriter.create<Torch::ConstantIntOp>(
+                binder.getLoc(), rewriter.getI64IntegerAttr(resultShape[i])));
+          }
+        }
+        Value outputSizeList = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            Torch::ListType::get(Torch::IntType::get(binder.op->getContext())),
+            outputSize);
+        Value outputTensor;
+        Type returnIndicesType = torch::Torch::ValueTensorType::get(
+            binder.op->getContext(), resultShape,
+            rewriter.getIntegerType(64, true));
+        if (inputRank == 3) {
```

Comment:
instead of running through the if statement and then returning, use early exits and just return failure at the end of the function

https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code